mesonbuild/backend/backends.py

```diff
@@ -572,17 +573,22 @@ def as_meson_exe_cmdline(self, exe: T.Union[str, mesonlib.File, build.BuildTarge
         cmd.extend(cmd_args)
         es = self.get_executable_serialisation(cmd, workdir, extra_bdeps, capture, feed, env, verbose=verbose)
         reasons: T.List[str] = []
+        serialize_with_env = shutil.which('env') is not None
         if es.extra_paths:
             reasons.append('to set PATH')
+            serialize_with_env = False
```

Comment:
Can't we use `env` to set `PATH` too?

Reply:
PATH is special, because we don't set it as an override but rather enhance the one that already exists. So running the ExecutableSerialisation will actually set the PATH to `extra_path:$PATH`.

This would either need to be baked in at configure time, or do what we currently do, which is calculate `os.environ['PATH']` at build time and append it.
Unfortunately, the env program isn't specialized enough to have an option that *prepends* (or appends) an environment variable to whatever its current value is. Usually people use shells for that.

Meson *could* use `sh -c 'env FOO=bar "PATH=/extra/path:$PATH" mycmd'` but at that point just use sh instead of env... Recursive sh invocations is really not a good rabbit hole to dive into (we already safe-quote arguments because ninja uses a shell to run every build edge).

Reply:
Fair enough.